tests/test_hydra.py

```diff
@@ -440,16 +444,77 @@ def test_cfg(tmpdir: Path, flag: str, expected_keys: List[str]) -> None:
         param(["--cfg=job", "--package=db.driver"], "mysql\n", id="package=db.driver"),
     ],
 )
-def test_cfg_with_package(tmpdir: Path, flags: List[str], expected: str) -> None:
+def test_cfg_with_package(
+    tmpdir: Path, flags: List[str], resolve: bool, expected: str
+) -> None:
     cmd = [
         "examples/tutorials/basic/your_first_hydra_app/5_defaults/my_app.py",
         "hydra.run.dir=" + str(tmpdir),
     ] + flags
+    if resolve:
+        cmd.append("--resolve")
 
     result, _err = run_python_script(cmd)
     assert normalize_newlines(result) == expected.rstrip()
 
 
+@mark.parametrize(
+    "resolve,expected",
+    [
+        param(
+            False,
+            dedent(
+                """\
+                node:
+                  loompa: 10
+                  zippity: ${node.loompa}
+                  do: oompa ${node.loompa}
+                  waldo: ???
+                """
+            ),
+            id="cfg",
+        ),
+        param(
+            True,
+            dedent(
+                """\
+                node:
+                  loompa: 10
+                  zippity: 10
+                  do: oompa 10
+                  waldo: ???
+                """
+            ),
+            id="resolve",
+        ),
+    ],
+)
+def test_cfg_resolve_interpolation(tmpdir: Path, resolve: bool, expected: str) -> None:
+    cmd = [
+        "examples/tutorials/basic/your_first_hydra_app/3_using_config/my_app.py",
+        "hydra.run.dir=" + str(tmpdir),
+        "--cfg=job",
+    ]
+    if resolve:
+        cmd.append("--resolve")
+
+    result, _err = run_python_script(cmd)
+    assert normalize_newlines(result) == expected.rstrip()
```

Comment:
there is a higher level output compare function that prints the diff on mismatch. I think you can use it here.

Reply:
Done in https://github.com/facebookresearch/hydra/pull/1616/commits/5b564c38a391b4fd36ed4deac3c09735188c28e0.